src/escalus_connection.erl

```diff
@@ -185,6 +192,14 @@ maybe_set_jid(Client = #client{props = Props}) ->
             Client
     end.
 
+-spec send_and_receive(client(), exml:element(), receive_options()) ->
```

Comment:
```suggestion
-spec send_and_receive(client(), exml:element(), receive_options()) -> R when
      R :: {exml_stream:element(), metadata()} |
           exml_stream:element() |
           {error, timeout}.
```

Reply:
This is just a style issue, but I think that if we need to break a type into multiple lines, it helps readability (thanks to indentation), to also give it a name.

Reply:
I will try to fix this with some other spec improvements that need to be done (e.g. `term()` used in too many places in `escalus_pred`).